src/codegen/codegen_acc_visitor.cpp

```diff
@@ -265,53 +275,57 @@ std::string CodegenAccVisitor::get_variable_device_pointer(const std::string& va
-                                                                                     variable);
+    return "nt->compute_gpu ? cnrn_target_deviceptr({}) : {}"_format(variable, variable);
 }
 
 
 void CodegenAccVisitor::print_newtonspace_transfer_to_device() const {
     int list_num = info.derivimplicit_list_num;
-    printer->add_line("if (nt->compute_gpu) {");
-    printer->add_line("    auto device_vec = static_cast<double*>(acc_copyin(vec, vec_size));");
-    printer->add_line("    auto device_ns = static_cast<NewtonSpace*>(acc_deviceptr(*ns));");
-    printer->add_line("    auto device_thread = static_cast<ThreadDatum*>(acc_deviceptr(thread));");
+    printer->start_block("if(nt->compute_gpu)");
+    printer->add_line("double* device_vec = cnrn_target_copyin(vec, vec_size / sizeof(double));");
+    printer->add_line("void* device_ns = cnrn_target_deviceptr(*ns);");
+    printer->add_line("ThreadDatum* device_thread = cnrn_target_deviceptr(thread);");
     printer->add_line(
-        "    acc_memcpy_to_device(&(device_thread[{}]._pvoid), &device_ns, sizeof(void*));"_format(
+        "cnrn_target_memcpy_to_device(&(device_thread[{}]._pvoid), &device_ns);"_format(
             info.thread_data_index - 1));
     printer->add_line(
-        "    acc_memcpy_to_device(&(device_thread[dith{}()].pval), &device_vec, sizeof(double*));"_format(
+        "cnrn_target_memcpy_to_device(&(device_thread[dith{}()].pval), &device_vec);"_format(
             list_num));
-    printer->add_line("}");
+    printer->end_block(1);
 }
 
 
 void CodegenAccVisitor::print_instance_variable_transfer_to_device() const {
-    printer->add_line("if (nt->compute_gpu) {");
-    printer->add_line("    auto dml = (Memb_list*) acc_deviceptr(ml);");
-    printer->add_line("    acc_memcpy_to_device(&(dml->instance), &inst, sizeof(void*));");
-    printer->add_line("}");
+    printer->start_block("if(nt->compute_gpu)");
+    printer->add_line("Memb_list* dml = cnrn_target_deviceptr(ml);");
+    printer->add_line("cnrn_target_memcpy_to_device(&(dml->instance), &(ml->instance));");
+    printer->end_block(1);
 }
 
 
 void CodegenAccVisitor::print_deriv_advance_flag_transfer_to_device() const {
-    printer->add_line("#pragma acc update device (deriv_advance_flag) if (nt->compute_gpu)");
+    printer->add_line("nrn_pragma_acc(update device (deriv_advance_flag) if(nt->compute_gpu))");
+    printer->add_line("nrn_pragma_omp(target update to(deriv_advance_flag) if(nt->compute_gpu))");
 }
 
 
 void CodegenAccVisitor::print_device_atomic_capture_annotation() const {
-    printer->add_line("#pragma acc atomic capture");
+    printer->add_line("nrn_pragma_acc(atomic capture)");
+    printer->add_line("nrn_pragma_omp(atomic capture)");
 }
 
 
 void CodegenAccVisitor::print_device_stream_wait() const {
-    printer->add_line("#pragma acc wait(nt->stream_id)");
+    printer->start_block("if(nt->compute_gpu)");
+    printer->add_line("nrn_pragma_acc(wait async(nt->streams[nt->stream_id]))");
```

Comment:
@iomaganaris / @olupton: does`acc wait async(stream_id)` change any execution semantics here compared to previous `acc wait (stream_id)` ? 

> from what I read in the nvforum - “wait async” is used to create dependencies between different async queues. So if your parallel regions had different queue numbers, you add “wait(Q1) async(Q2)” where Q2 should wait for Q1 to finish before running.